daxgen/daxgen.py

```diff
@@ -77,19 +79,35 @@ def write(self, filename, name='dax'):
         # Add files to DAX-level replica catalog.
         catalog = {}
         for v in files:
-            lfn, pfns = self.graph.node[v]['lfn'], self.graph.node[v]['pfns']
-            f = File(lfn)
-            for pfn, site in pfns:
-                f.addPFN(pfn, site)
-            catalog[lfn] = f
-        for f in catalog.values():
+            attrs = self.graph.node[v]
+            f = File(attrs['lfn'])
+
+            # Add physical file names, if any.
+            urls = attrs.get('urls')
+            if urls is not None:
+                sites = attrs.get('sites')
+                if sites is None:
+                    sites = ','.join(len(urls) * ['local'])
+                for url, site in zip(urls.split(','), sites.split(',')):
+                    f.addPFN(PFN(url, site))
+
+            catalog[attrs['lfn']] = f
             dax.addFile(f)
 
         # Add jobs to the DAX.
         for v in tasks:
             attrs = self.graph.node[v]
             job = Job(name=attrs['name'], id=v)
-            job.addArguments(attrs['args'].split())
+
+            # Add job command line arguments replacing any file name with
+            # respective Pegasus file object.
+            args = attrs['args'].split()
+            lfns = list(set(catalog) & set(args))
+            if lfns:
```

Comment:
I meant `lfns` and later `args` as well. not sure when args might be empty but it might be worth to check for it